BEIMA.Backend.Test/DeviceFunctions/DeleteDeviceTest.cs

```diff
@@ -17,17 +20,21 @@ public class DeleteDeviceTest : UnitTestBase
         [TestCase("xxx")]
         [TestCase("123")]
         [TestCase("jddkkslo9402mdjgkflsnxjf")]
-        public void IdIsInvalid_DeleteDevice_ReturnsInvalidId(string id)
+        public async Task IdIsInvalid_DeleteDevice_ReturnsInvalidId(string id)
         {
             // ARRANGE
             Mock<IMongoConnector> mockDb = new Mock<IMongoConnector>();
             MongoDefinition.MongoInstance = mockDb.Object;
 
+            // Setup storage provider.
+            Mock<IStorageProvider> mockStorage = new Mock<IStorageProvider>();
+            var storageProvider = mockStorage.Object;
+
             var request = CreateHttpRequest(RequestMethod.POST);
             var logger = (new LoggerFactory()).CreateLogger("Testing");
 
             // ACT
-            var response = DeleteDevice.Run(request, id, logger);
+            var response = await new DeleteDevice(storageProvider).Run(request, id, logger);
 
             // ASSERT
             Assert.DoesNotThrow(() => mockDb.Verify(mock => mock.DeleteDevice(It.IsAny<ObjectId>()), Times.Never));
```

Comment:
You should also make sure to add an assert here to make sure that the mockStorage didn't get called. (You can look at this mockDb.Verify as an example.)